Lets Do This/Controllers/Settings/LDTSettingsViewController.m

```diff
@@ -157,13 +157,14 @@ - (void)handleLogoutTap:(UITapGestureRecognizer *)recognizer {
 
         [[DSOUserManager sharedInstance] endSessionWithCompletionHandler:^ {
             [SVProgressHUD dismiss];
-            [self.navigationController pushViewController:[[LDTUserConnectViewController alloc] init] animated:YES];
-            [self.navigationController styleNavigationBar:LDTNavigationBarStyleClear];
-            // Now that tabBar is hidden, select the first tab, so it will be the first tab selected upon next login.
-            LDTTabBarController *tabBar = (LDTTabBarController *)[[[[UIApplication sharedApplication] delegate] window] rootViewController];
-            [tabBar setSelectedIndex:0];
+            [self presentUserConnectViewController];
         } errorHandler:^(NSError *error) {
             [SVProgressHUD dismiss];
+            // Performs normal logout functionality even if we are presented with an error,
+            // but only if error is not a lack of connectivity. 
+            if (error.code != -1009) {
+                [self presentUserConnectViewController];
+            }
             [LDTMessage displayErrorMessageForError:error];
```

Comment:
I think this needs to be inside an `else`. Otherwise, in the instance where we get a "No active sessions" message.. this will still display the error message, which will be confusing to the user.